core/Archive/ArchivePurger.php

```diff
@@ -166,9 +198,11 @@ protected function deleteArchiveIds(Date $date, $idArchivesToDelete)
         $numericTable = ArchiveTableCreator::getNumericTable($date);
         $blobTable    = ArchiveTableCreator::getBlobTable($date);
 
+        $deletedCount = 0;
         foreach ($batches as $idsToDelete) {
-            $this->model->deleteArchiveIds($numericTable, $blobTable, $idsToDelete);
+            $deletedCount += $this->model->deleteArchiveIds($numericTable, $blobTable, $idsToDelete);
         }
+        return $deletedCount;
```

Comment:
Do you mind adding an integration test for this? Would be nice to have a test for this. Not only for the return value but also whether it actually deletes the archives.